doc/source/development/extending.rst

```diff
@@ -208,13 +208,201 @@ will
 2. call ``result = op(values, ExtensionArray)``
 3. re-box the result in a ``Series``
 
+:class:`~pandas.api.extensions.ExtensionArray` Series Operations Support
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 0.25.0
+
+In addition to operators like `__mul__` and `__add__`, the pandas Series
+namespace provides a long list of useful operations such as :meth:`Series.round`,
+:meth:`Series.sum`, :meth:`Series.abs`, etc'. Some of these are handled by
+pandas own algorithm implementations (via a dispatch function), while others
+simply call an equivalent numpy function with data from the underlying array.
+In order to support this operations in a new ExtensionArray, you must provide
+an implementation for them.
+
+.. note::
+
+    There is a third category of operation which live on the `pandas`
+    namespace, for example `:meth:pd.concat`. There is an equivalent numpy
+    function `:meth:np.concatenate`, but it is not called by the pandas
+    method. In general, these function should just work with your EA, you do
+    not need to implement more than the general EA interface.
+
+As of 0.25.0, pandas provides its own implementations for the following
+operations: :meth:`Series.any`, :meth:`Series.all`,
+:meth:`Series.min`, :meth:`Series.max`, :meth:`Series.sum`,
+:meth:`Series.mean`, :meth:`Series.median`, :meth:`Series.prod`
+(and its alias :meth:`Series.product`), :meth:`Series.std`,
+:meth:`Series.var`, :meth:`Series.sem`,
+:meth:`Series.kurt`, and :meth:`Series.skew`.
+
+In order to implement any of these functions, your ExtensionArray must include
+an implementation of :meth:`ExtensionArray._reduce`. Once you provide an
+implementation of :meth:`ExtensionArray._reduce` which handles a particular
+method, calling the method on the Series will invoke the implementation
+on your ExtensionArray. All these methods are reduction functions, and
+so are expected to return a scalr value of some type. However it is perfectly
+acceptable to return some instance of an :class:`pandas.api.extensions.ExtensionArray`.
```

Comment:
When is returning an EA OK?

Reply:
What are you actually asking? I was trying to make it clear that overriding a numpy function doesn't mean you have to return a numpy array.

if it is appropriate to return an array-like (`np.round()` yes, `series.sum()` no), it's always appropriate to return an EA isn't it?

So far, whenever an EA override I wrote returned an EA instance the user got back an EA Series, as expected. 
It makes sense for an EA instance asked to  execute an operation on itself, to return a modified EA of the same type.

What's the problem you have with this?

Reply:
The question is about returning an array at all. This are reducing functions, so are they not expected to return a scalar?

Yes, for other methods an EA is appropriate. But this section is about the `_reduce`, which is only for reductions.